src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -0,0 +1,489 @@
+            Assert.True(prop.PartialImplementationPart!.GetPublicSymbol().IsExtern);
+        }
+
+        [Fact]
+        public void Extern_02()
+        {
+            var source = """
+                partial class C
+                {
+                    extern partial int P { get; set; }
+                    extern partial int P { get; set; }
+                }
+                """;
+            var comp = CreateCompilation([source, IsExternalInitTypeDefinition]);
+            comp.VerifyEmitDiagnostics(
+                // (3,24): error CS9301: Partial property 'C.P' must have an definition part.
+                //     extern partial int P { get; set; }
+                Diagnostic(ErrorCode.ERR_PartialPropertyMissingDefinition, "P").WithArguments("C.P").WithLocation(3, 24),
+                // (4,24): error CS9303: A partial property may not have multiple implementing declarations
+                //     extern partial int P { get; set; }
+                Diagnostic(ErrorCode.ERR_PartialPropertyDuplicateImplementation, "P").WithLocation(4, 24),
+                // (4,24): error CS0102: The type 'C' already contains a definition for 'P'
+                //     extern partial int P { get; set; }
+                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "P").WithArguments("C", "P").WithLocation(4, 24)
+                );
+        }
+
+        [Fact]
+        public void Semantics_01()
+        {
+            // happy definition + implementation case
+            var source = """
+                using System;
+
+                var c = new C { P = 1 };
+                Console.Write(c.P);
+
+                partial class C
+                {
+                    public partial int P { get; set; }
+                }
+
+                partial class C
+                {
+                    private int _p;
+                    public partial int P { get => _p; set => _p = value; }
+                }
+                """;
+            var verifier = CompileAndVerify(source, expectedOutput: "1");
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("C.P.get", """
+                {
+                  // Code size        7 (0x7)
+                  .maxstack  1
+                  IL_0000:  ldarg.0
+                  IL_0001:  ldfld      "int C._p"
+                  IL_0006:  ret
+                }
+                """);
+            verifier.VerifyIL("C.P.set", """
+                {
+                  // Code size        8 (0x8)
+                  .maxstack  2
+                  IL_0000:  ldarg.0
+                  IL_0001:  ldarg.1
+                  IL_0002:  stfld      "int C._p"
+                  IL_0007:  ret
+                }
+                """);
+
+            var comp = (CSharpCompilation)verifier.Compilation;
+            var cClass = comp.GetMember<NamedTypeSymbol>("C");
+            var members = cClass.GetMembers().SelectAsArray(m => m.ToTestDisplayString());
```

Comment:
Let's add some tests around `PartialDefinitionPart` and `PartialImplementationPart` for these success cases, namely verifying that they navigate correctly and return null correctly, as appropriate.